File: Add_Binary.java
/*
* Leetcode Challenge: Add Binary
*
* Given two binary strings a and b, return their sum as a binary string.
*
* 4/29/24
* Mark Robuck
*/
import java.math.BigInteger;

class Solution {
    /*
    *Sum two binary strings.
    * @parmas a, first binary string to add
    * @params b, second binary string to add
    * @return binary sum of a and b
    */
    public String addBinary(String a, String b) {
        //convert strings into binary numbers
        //in order to fix a runtime error for large values convert int to BigInteger
        BigInteger binaryA = new BigInteger(a, 2);
        BigInteger binaryB = new BigInteger(b, 2);
        //add binary numbers
        BigInteger binarySum = binaryA.add(binaryB);
        //convert binary back to string representation
        String sum = binarySum.toString(2);
        return sum;
    }
}

File: Add_Binary.py
"""
Leetcode Challenge: Add Binary

Given two binary strings a and b, return their sum as a binary string.

4/29/24
Mark Robuck
"""

class Solution:
    """
    Sum two binary strings.
    Args:
        a(str): first binary string to add
        b(str): second binary string to add
    Returns:
        str: binary sum of a and b
    """
    def addBinary(self, a: str, b: str) -> str:
        #convert strings to binary using integer casting of base 2
        binaryA = int(a, 2)
        binaryB = int(b, 2)
        #use binary addition and convert back to string
        binarySum = bin(binaryA + binaryB)
        #strip the "0b" portion from the front of the string
        result = binarySum[2:]
        return result
